### DISCUSSION TITLE
Pagination for table widget

### DISCUSSION THREAD

**Discussion by giacomomasseron** [filamentphp/filament#14375] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2.113

### How can we help you?

Hi,  
I have a table widget done like that:

	<?php

	namespace App\Filament\Clusters\Legacy\Widgets;

	use App\Models\Sushi\Warning;
	// ...

	class Warnings extends BaseWidget
	{
		protected int|string|array $columnSpan = 'full';

		public function table(Table $table): Table
		{
			return $table
				->query(Warning::version('V2'))
				->deferLoading()
				->columns([                
				])
				->filters([
				])
				->actions([
				]);
		}

		public static function canView(): bool
		{
			return true;
		}
	}

It works but the pagination is like the attachment
![pagination](https://github.com/user-attachments/assets/af4b8574-34cf-4032-a9c6-c38766e776bd)

I need to have pagination with numbers of records and pages number. How can I do that?

P.S. all the other resources of the same projects have the pagination in that way, it's just this widget.

**Comment by leandrocfe** [comment, marked as answer]
Add to your widget

```php
use Tables\Concerns\CanPaginateRecords;

public function table(Table $table): Table
{
...
```
